service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureNetworkInterfaceStepTest.java

```diff
@@ -0,0 +1,125 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.vm;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+import bio.terra.landingzone.library.landingzones.deployment.SubnetResourcePurpose;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneApiDispatch;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneDeployedResource;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.network.ControlledAzureNetworkResource;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.resourcemanager.network.NetworkManager;
+import com.azure.resourcemanager.network.models.Network;
+import com.azure.resourcemanager.network.models.Networks;
+import com.azure.resourcemanager.network.models.Subnet;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("unit")
+public class CreateAzureNetworkInterfaceStepTest extends BaseUnitTest {
```

Comment:
Isn't this an Azure test? Should it be derived from BaseAzureTest?
If it is doing anything that requires the azure-enabled feature to be true, then it should be an Azure test.

Even though azure-enabled is true for all Broad environments it is not true for all environments.

Reply:
Yeah, I initially had it like that, however I noticed the base test sets a Spring profile for Azure, which I don't require for this test. It is not clear to me what is the best way to describe a unit test that applies to Azure if I don't require a spring context. Happy to change it to BaseAzureTest is that is the best answer we have right now, please let me know.